lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1174,9 +1181,69 @@ sub delete_logs {
         path($result_dir, 'ulogs')->list_tree({hidden => 1}),
         find_video_files($result_dir),
     );
-    my $deleted_size = 0;
-    $deleted_size += $_->reduce(sub { $a + _delete_returning_size($b) }, 0) for @files;
+    my $deleted_size = _delete_returning_size_from_array(\@files);
     $self->update({logs_present => 0, result_size => \"result_size - $deleted_size"});
+    return $deleted_size;
+}
+
+sub delete_videos {
+    my ($self) = @_;
+
+    my $result_dir = $self->result_dir;
+    return 0 unless $result_dir;
+
+    my @files        = (find_video_files($result_dir), Mojo::Collection->new(path($result_dir, 'video_time.vtt')));
+    my $deleted_size = _delete_returning_size_from_array(\@files);
+    $self->update({result_size => \"result_size - $deleted_size"});    # considering logs still present here
+    return $deleted_size;
+}
+
+sub delete_results {
+    my ($self) = @_;
+
+    # delete the entire results directory
+    if (my $result_dir = $self->result_dir) {
+        File::Path::rmtree($result_dir) if -d $result_dir;
+    }
+
+    # delete all screenshot links and all exclusively used screenshots
+    my $job_id                          = $self->id;
+    my $exclusively_used_screenshot_ids = $self->exclusively_used_screenshot_ids;
+    my $schema                          = $self->result_source->schema;
+    my $screenshots                     = $schema->resultset('Screenshots');
+    my $dbh                             = $schema->storage->dbh;
+    my $delete_screenshot_query         = $dbh->prepare('DELETE FROM screenshots WHERE id = ?');
+    my $imagesdir                       = imagesdir();
+    $self->screenshot_links->delete;
+    for my $screenshot_id (@$exclusively_used_screenshot_ids) {
+        # get rid of screenshot, similar to OpenQA::Task::Job::Limit
```

Comment:
Why do we need to do this here? Existing code seems to assume that `screenshots left unused are deleted later in the job limit task`.

Reply:
We need to delete screenshots here immediately because otherwise we would not be able to see a difference immediately when re-checking df again.

Reply:
Right, that makes sense in hindsight. Thanks for clarifying!